tables/endpointsecurity/src/fileeventstableplugin.cpp

```diff
@@ -0,0 +1,171 @@
+    { "cdhash", IVirtualTable::ColumnType::String },
+    { "path", IVirtualTable::ColumnType::String },
+    { "file_path", IVirtualTable::ColumnType::String }
+  };
+  // clang-format on
+
+  return kTableSchema;
+}
+
+Status FileEventsTablePlugin::generateRowList(RowList &row_list) {
+  std::lock_guard<std::mutex> lock(d->row_list_mutex);
+
+  row_list = std::move(d->row_list);
+  d->row_list = {};
+
+  return Status::success();
+}
+
+Status FileEventsTablePlugin::processEvents(
+    const IEndpointSecurityConsumer::EventList &event_list) {
+  RowList generated_row_list;
+
+  for (const auto &event : event_list) {
+    Row row;
+
+    auto status = generateRow(row, event);
+    if (!status.succeeded()) {
+      return status;
+    }
+
+    if (!row.empty()) {
+      generated_row_list.push_back(std::move(row));
+    }
+  }
+
+  {
+    std::lock_guard<std::mutex> lock(d->row_list_mutex);
+
+    // clang-format off
+    d->row_list.insert(
+      d->row_list.end(),
+      std::make_move_iterator(generated_row_list.begin()),
+      std::make_move_iterator(generated_row_list.end())
+    );
+    // clang-format on
+
+    if (d->row_list.size() > d->max_queued_row_count) {
+      auto rows_to_remove = d->row_list.size() - d->max_queued_row_count;
+
+      d->logger.logMessage(IZeekLogger::Severity::Warning,
+                           "file_events: Dropping " +
+                               std::to_string(rows_to_remove) +
+                               " rows (max row count is set to " +
+                               std::to_string(d->max_queued_row_count));
+
+      // clang-format off
+      d->row_list.erase(
+        d->row_list.begin(),
+        std::next(d->row_list.begin(), rows_to_remove)
+      );
+      // clang-format on
+    }
+  }
+
+  return Status::success();
+}
+
+FileEventsTablePlugin::FileEventsTablePlugin(IZeekConfiguration &configuration,
+                                             IZeekLogger &logger)
+    : d(new PrivateData(configuration, logger)) {
+
+  d->max_queued_row_count = d->configuration.maxQueuedRowCount();
+}
+
+Status FileEventsTablePlugin::generateRow(
+    Row &row, const IEndpointSecurityConsumer::Event &event) {
+
+  row = {};
+
+  std::string action;
+
+  switch (event.type) {
+  case IEndpointSecurityConsumer::Event::Type::Open:
+    action = "open";
+    break;
+  case IEndpointSecurityConsumer::Event::Type::Create:
+    action = "create";
+    break;
+
+  default:
+    return Status::success();
+  }
+  const auto &header = event.header;
+  row["timestamp"] = static_cast<std::int64_t>(header.timestamp);
+  row["parent_process_id"] =
+      static_cast<std::int64_t>(header.parent_process_id);
+  row["orig_parent_process_id"] =
+      static_cast<std::int64_t>(header.orig_parent_process_id);
+  row["process_id"] = static_cast<std::int64_t>(header.process_id);
+  row["user_id"] = static_cast<std::int64_t>(header.user_id);
+  row["group_id"] = static_cast<std::int64_t>(header.group_id);
```

Comment:
These `uint64_t` to `int64_t` conversions could in principle overflow (`timestamp` caught my eye here). WDYT about e.g., explicitly `assert`ing that the values fit in the `int64_t` range?

Reply:
I have added an assertion in my recent commit to check the range of `timestamp` only. All the other casts are from int32 so technically they should not overflow. right?